src/main/kotlin/com/exactpro/th2/rptdataprovider/services/cradle/MessageGroupCradleService.kt

```diff
@@ -70,13 +71,39 @@ class MessageGroupCradleService(
         // CradleStorage.getMessageCountersAsync add FrameType.getMillisInFrame to get the last next lest interval
         // Cradle should support null value for from / to Interval properties
         private val CRADLE_MAX_TIMESTAMP: Instant = Instant.ofEpochMilli(Long.MAX_VALUE).minusMillis(FrameType.values().asSequence().map(FrameType::getMillisInFrame).maxOf(Long::toLong))
+
+        // source interval changes to exclude intersection to next or previous page
+        private fun PageInfo.toInterval(): Interval = Interval(
+            started?.plusNanos(1) ?: CRADLE_MIN_TIMESTAMP,
+            ended?.minusNanos(1) ?: CRADLE_MAX_TIMESTAMP
```

Comment:
Shouldn't we change only one boundary?

Reply:
cradle API includes start and end page during search